tests/filecheck/dialects/emitc/emitc_types.mlir

```diff
@@ -0,0 +1,14 @@
+// RUN: XDSL_ROUNDTRIP
+
+// CHECK: f32_2D = !emitc.array<4x2xf32>
+// CHECK-SAME: i32_1D = !emitc.array<10xi32>
+// CHECK-SAME: f64_1D = !emitc.array<5xf64>
+// CHECK-SAME: i1_3D = !emitc.array<3x4x5xi1>
+// CHECK-SAME: EmitC_ArrayType shape must not be empty
```

Comment:
I would recommend adding a separate test for this, something like `tests/filecheck/dialects/emitc/emitc_types_invalid.mlir`, and passing `--verify-diagnostics`, you can do a search in the project for similar tests.

Reply:
I think we need to yoink this line now and then we'll be gtg